src/region/RegionCreator.tsx

```diff
@@ -188,13 +204,16 @@ export default class RegionCreator extends React.Component<Props, State> {
             : {};
 
         return (
-            <div
+            <AutoScroller
                 ref={this.creatorRef}
-                className={classNames(className, 'ba-RegionCreator', {
-                    'is-active': canDraw,
-                })}
+                as="svg"
+                className={classNames(className, 'ba-RegionCreator', { 'is-active': canDraw })}
+                enabled={isDrawing}
+                onScroll={this.handleScroll}
                 {...eventHandlers}
-            />
+            >
+                <RegionRect shape={this.getShape()} />
```

Comment:
Does the fact that `RegionRect` returns null if the shape prop is falsy matter? Should we check the shape before even rendering `AutoScroller`?

Reply:
The root svg element is doing a lot of conditional work here. We need event handling for drawing (sometimes), the autoscroll behavior (if the user is actively drawing), and the actual. shape (always, if it exists). I don't think I want to make things more complicated without good reason. The main performance hit comes while the user is actively drawing, which means there's a shape present.

Reply:
👍